### ISSUE TITLE
Randomness in LayerConductance With Non-Linear Layers?

### ISSUE THREAD

**Issue by jlmitche23** [pytorch/captum#830]
Hi,

I'm running `LayerConductance` on two models:  One with the following architecture: 

```
MNIST(
  (fc1): Linear(in_features=784, out_features=100, bias=True)
  (relu1): ReLU()
  (fc2): Linear(in_features=100, out_features=100, bias=True)
  (relu2): ReLU()
  (fc3): Linear(in_features=100, out_features=10, bias=True)
)
```

And the other is Squeezenet1_1, loaded from `torchvision`.

When I run `attribute` on the MNIST network's fully connected layers given a fixed input, I get deterministic results each time, without setting a random seed.  However, when I do the same for a layer in Squeezenet (pooling or convolution), I fail to get deterministic results.  The results do not preserve relative magnitude (i.e. in one run a neuron could be more positively important than another, and in a second run, the relationship reverses).

I do get determinist results, when setting a seed as expected.

Where does the randomness in this case come from?  It was not immediately apparent to me after a brief look through the source code.  What guarantees do we have about the conductance values returned?  Will neurons with a conductance value of 0 always remain 0 regardless of the seed?

Thank you in advance!

**Comment by vivekmig** [comment]
Hi @jlmitche23 , conductance itself should return deterministic results as long as the model's forward / backward passes are also deterministic. I suspect that dropout may be enabled in your Squeezenet model, which could be causing this issue.

Can you try running a forward pass and verify the result is consistent when repeated for the same inputs? If not, you can try setting `model.eval()` before computing attributions to ensure the model is in eval-mode and dropout is disabled.

**Comment by jlmitche23** [comment]
Yes, I just tried setting the model to eval-mode and that did the trick.  Thank you so much!
